src/Ensconce.Cli/ApplicationInteraction.cs

```diff
@@ -15,66 +16,100 @@ internal static class ApplicationInteraction
     {
         internal static void StopAndDeleteServicesInDirectory(string directory)
         {
-            foreach (var serviceDetails in GetServicesInstalledInDirectory(directory))
+            Logging.Log("Finding services to stop and delete in {0}", directory);
+            var services = GetServicesInstalledInDirectory(directory).ToList();
+            if (services.Any())
             {
-                var service = ServiceController.GetServices().First(svc => svc.DisplayName == serviceDetails.DisplayName);
-                if (service.Status != ServiceControllerStatus.Stopped)
+                Logging.Log("{0} services found in {1}", services.Count, directory);
+                foreach (var serviceDetails in services)
                 {
-                    Logging.Log("Stopping service {0}", serviceDetails.DisplayName);
-                    service.Stop();
-                }
+                    var service = ServiceController.GetServices().First(svc => svc.DisplayName == serviceDetails.DisplayName);
```

Comment:
Should probably cache this too:

`ServiceController.GetServices()`

Reply:
The reason not too is because if 2 services are linked, stopping 1 would stop the other and therefore you want to go get the services again to get the correct status.